src/bcrhp/bcrhp/migrations/947_historic_site_one_line.py

```diff
@@ -0,0 +1,356 @@
+                pid parcel_id,
+                street_address,
+                city,
+                province,
+                locality,
+                location_description,
+                authorities->>'government_name' government_name,
+                authorities->>'authority' government_level,
+                authorities->>'recognition_type' protection_type,
+                to_date(authorities->>'start_date', 'YYYY-MM-DD') protection_start_date,
+                authorities->>'reference_number' reference_number,
+                case when chronology is null then null else to_char(to_date(chronology[0]->>'start_year', 'YYYY-MM-DD'), 'YYYY')::numeric(4,0) end construction_date,
+                case when chronology is not null and (chronology[0]->>'dates_approximate')::boolean then 'Circa' end construction_date_qualifier,
+                case when significance_statement is null then null else (significance_statement->>'significance_type') end significance_type,
+                case when significance_statement is null then null else (significance_statement->>'physical_description') end physical_description,
+                case when significance_statement is null then null else (significance_statement->>'heritage_value') end heritage_value,
+                case when significance_statement is null then null else (significance_statement->>'defining_elements') end defining_elements,
+                case when significance_statement is null then null else (significance_statement->>'document_location') end document_location,
+                actors->>'Architect / Designer' architect_name,
+                actors->>'Builder' builder_name,
+                functional_states->>'Current' current_function,
+                functional_states->>'Historic' historic_function,
+                dimensions_area_sqm,
+                gis_latitude,
+                gis_longitude,
+                utm_zone,
+                utm_northing,
+                utm_easting,
+                site_url,
+                site_boundary
+            from V_HISTORIC_SITE
+            where bcrhp_submission_status in ('Approved - Basic Record','Approved - Full Record')
+                and registration_status in ('Federal Jurisdiction', 'Recorded/Unprotected', 'Registered', 'Legacy')
+                and exists (
+                    select 1
+                    from jsonb_array_elements(chronology) elem
+                    where elem->>'event' = 'Construction'
+                )
+                and not coalesce(restricted, false)
+            ;
+            """,
+            """
+            drop view if exists databc.V_HISTORIC_ENVIRO_ONEROW_SITE cascade;
+            """),
+        RunPrivilegedSQL(
+             """
+            create or replace view heritage_site.csv_export as
+            select
+                resourceinstanceid as site_id
+                , borden_number as "Borden Number"
+                , street_address as "Street Address"
```

Comment:
For rows that can have a null value, please replace null value with empty string. Null values come out as "None" in the csv.